dotCMS/src/main/java/com/dotcms/http/CircuitBreakerUrl.java

```diff
@@ -192,12 +217,15 @@ public void setWriteListener(WriteListener writeListener) {
         });
     }
 
-    
-    
-    
-    
-    
     public void doOut(final HttpServletResponse response) throws IOException {
+
+        if (threadIdConnectionCountSet.size() >= circuitBreakerMaxConnTotal.get()) {
+
+            Logger.info(this, "The maximum number of connections has been reached, size: " + threadIdConnectionCountSet.size());
+            throw new RejectedExecutionException("The maximum number of connections has been reached.");
+        }
+
+        threadIdConnectionCountSet.add(Thread.currentThread().getId());
```

Comment:
Does it make sense to add the current thread in the try?

Reply:
also, under load, it might be possible to exceed the `circuitBreakerMaxConnTotal` by a few.  This might be acceptable rather than syncing